test/lit/passes/type-merging.wast

```diff
@@ -277,15 +416,93 @@
   )
 )
 
-;; Check that a ref.test inhibits merging (ref.cast is already checked above).
+;; Function types
 (module
-  ;; CHECK:      (type $ref|$A|_=>_i32 (func (param (ref $A)) (result i32)))
+  ;; CHECK:      (rec
+  ;; CHECK-NEXT:  (type $func (func (param eqref)))
+  (type $func (func (param eqref)))
+  (type $sub-func (func_subtype (param eqref) $func))
+  ;; CHECK:       (type $sub-func-refined (func_subtype (param anyref) $func))
+  (type $sub-func-refined (func_subtype (param anyref) $func))
+
+  ;; CHECK:       (type $none_=>_none (func))
 
-  ;; CHECK:      (type $A (struct ))
+  ;; CHECK:      (func $foo (type $none_=>_none)
+  ;; CHECK-NEXT:  (local $a (ref null $func))
+  ;; CHECK-NEXT:  (local $b (ref null $func))
+  ;; CHECK-NEXT:  (local $c (ref null $sub-func-refined))
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT: )
+  (func $foo
+    ;; $func will remain the same.
+    (local $a (ref null $func))
+    ;; $sub-func will be merged into $func.
+    (local $b (ref null $sub-func))
+    ;; $sub-func-refined will not be merged into $func because it refines a result.
+    (local $c (ref null $sub-func-refined))
+  )
+)
+
+;; Check that public types are not merged.
+;; TODO: $C still appears in the output because type-updating.cpp does not
+;; support merging private types into their public supertypes very well. Fix
+;; this.
```

Comment:
Is this TODO done? I don't see `$C` in the output.

Reply:
Ah yes, this was a note I added last night to remind me of what was happening, but it's resolved now.